app/models/container.rb

```diff
@@ -30,8 +30,8 @@ def event_where_clause(assoc = :ems_events)
     case assoc.to_sym
     when :ems_events
       # TODO: improve relationship using the id
-      ["container_namespace = ? AND #{events_table_name(assoc)}.ems_id = ?", container_project.name,
-       ext_management_system.id]
+      ["container_namespace = ? AND #{events_table_name(assoc)}.ems_id = ? AND container_name = ?",
+       container_project.name, ext_management_system.id, name]
```

Comment:
> is it possible to use container_id?
> I see it in the table, but maybe we still leave that empty

It is possible, but currently all other entities with events work with entity name.
There is a TODO referring to this option, but for this change IMHO we should maintain consistency and discuss if we want to change that for all entities in a separate PR.
@simon3z what do you think?